libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1074,25 +1074,39 @@ bool TypeChecker::visit(Conditional const& _conditional)
 
 	TypePointer trueType = type(_conditional.trueExpression())->mobileType();
 	TypePointer falseType = type(_conditional.falseExpression())->mobileType();
+
+	TypePointer commonType;
+
 	if (!trueType)
-		m_errorReporter.fatalTypeError(_conditional.trueExpression().location(), "Invalid mobile type.");
+		m_errorReporter.typeError(_conditional.trueExpression().location(), "Invalid mobile type in true expression.");
+	else
+		commonType = trueType;
+
 	if (!falseType)
-		m_errorReporter.fatalTypeError(_conditional.falseExpression().location(), "Invalid mobile type.");
+		m_errorReporter.typeError(_conditional.falseExpression().location(), "Invalid mobile type in false expression.");
+	else
+		commonType = falseType;
 
-	TypePointer commonType = Type::commonType(trueType, falseType);
-	if (!commonType)
+	if (!trueType && !falseType)
```

Comment:
Isn't this equivalent to `!commonType`?

Reply:
It is. not sure it's more clear if I change it though.

Reply:
It is not, you could have `true ? true : 7`, then commonType would be nullptr.

Reply:
It's not equivalent, you can have `true ? false : 7`, then commonType would still be nullptr, but trueType and falseFalse would be valid types.

Reply:
Ah sorry, did not realize the line is deleted.